packer/provision/local-docker.yaml

```diff
@@ -2,3 +2,215 @@
+        ["qemu-kvm", "virt-top", "libguestfs-tools", "virtinst", "bridge-utils"]
+
+    - name: Enable vhost_net
+      community.general.modprobe:
+        name: vhost_net
+        state: present
+
+    - name: Add vhost_net to kernel modules
+      shell: echo "vhost_net" | sudo tee -a /etc/modules
+      args:
+        executable: /bin/bash
+
+    - name: Disable kvm_intel
+      community.general.modprobe:
+        name: kvm_intel
+        state: absent
+
+    - name: Add nested option to kvm-intel
+      shell: echo 'options kvm-intel nested=y' >> /etc/modprobe.d/dist.conf
+      args:
+        executable: /bin/bash
+
+    - name: Enable kvm_intel
+      community.general.modprobe:
+        name: kvm_intel
+        state: present
+
+    - name: Restart libvirtd
+      service:
+        name: libvirtd
+        state: restarted
+
+    - name: Ensure groups exist
+      ansible.builtin.group:
+        name: "{item}"
+        state: present
+      loop:
+        - sudo
+        - kvm
+        - libvirt
+        - docker
+
+    - name: set up jenkins user
+      user:
+        name: jenkins
+        groups: sudo,kvm,libvirt
+
+    - name: Create a directory if it does not exist
+      file:
+        path: /var/jenkins/
+        state: directory
+        owner: jenkins
+        group: jenkins
+        mode: "0755"
+
+    - name: Add ansible repo
+      apt_repository:
+        repo: ppa:ansible/ansible
+        state: present
+
+    - name: Install ansible
+      apt: update_cache=yes name={{ item }} state=latest
+      loop: ["ansible"]
+
+    - name: Unarchive a file that needs to be downloaded (added in 2.0)
+      unarchive:
+        src: https://golang.org/dl/go1.15.2.linux-amd64.tar.gz
+        dest: /usr/local/
+        remote_src: yes
+
+    - name: Link go binary
+      file:
+        src: /usr/local/go/bin/go
+        dest: /usr/local/bin/go
+        state: link
+
+    - name: Link gofmt binary
+      file:
+        src: /usr/local/go/bin/gofmt
+        dest: /usr/local/bin/gofmt
+        state: link
+
+    - name: install dependencies
+      apt: update_cache=yes name={{ item }} state=latest
+      loop:
+        [
+          "make",
+          "libssl-dev",
+          "zlib1g-dev",
+          "libbz2-dev",
+          "libreadline-dev",
+          "libsqlite3-dev",
+          "wget",
+          "curl",
+          "llvm",
+          "libncurses5-dev",
+          "libncursesw5-dev",
+          "xz-utils",
+          "tk-dev",
+          "libffi-dev",
+          "liblzma-dev",
+          "python-openssl",
+          "git",
+          "unzip",
+          "jq",
+        ]
+
+    - name: Download pyenv
+      git:
+        repo: "https://github.com/pyenv/pyenv.git"
+        dest: /home/jenkins/.pyenv
+
+    - name: Link pyenv binary
+      file:
+        src: /home/jenkins/.pyenv/bin/pyenv
+        dest: /usr/local/bin/pyenv
+        state: link
+
+    - name: Install python 3.7.0
+      shell: pyenv install 3.7.0 -f
+      become: jenkins
```

Comment:
When testing this manually, I got the following error while trying to install the tools as jenkins

    openstack: the field 'become' has an invalid value (jenkins), and could not be converted to an bool.The error was: The value 'jenkins' is not a valid boolean.  Valid booleans include: 't', 1, '0', 0, 'off', 'no', 'true', 'y', 'false', 'f', 'on', 'yes', '1', 'n'. The value 'jenkins' is not a valid boolean.  Valid booleans include: 't', 1, '0', 0, 'off', 'no', 'true', 'y', 'false', 'f', 'on', 'yes', '1', 'n'

I ended up running all these with the global "become: true"

Reply:
recheck

Reply:
Changed it to become_user